### DISCUSSION TITLE
Welcome to oatbar Discussions!

### DISCUSSION THREAD

**Discussion by igor-petruk** [igor-petruk/oatbar#13] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas of what could be done better, especially around the config language and quality-of-life improvements
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Dan7h3x** [comment]
Hi, first thank you its a good and capable status bar,
my advice and request is to put a example config file with preview of it in documentation for easy use in first try, and have a systray support for the oatbar at first place, i hope this project grows by good and be more stable, again thank you.

**Reply by igor-petruk** [reply]
Hi thank you for kind words :)

The documentation is extensive and the config generated by default provides a good starting point. To improve this I was thinking of either

- Having a long config with all features demonstrated as you suggest.
- Build a CLI tool which would add entries step by step by asking questions to quickly ramp up the initial config

**Reply by igor-petruk** [reply]
Another cool idea I have about the CLI setup wizard is to create variable for a theme and let color picker dialog dynamically spin, streaming the variable into a bar. It should let the user to tune their colors in real-time

**Reply by Dan7h3x** [reply]
Yeah they are good ideas.
for me, since i'm using dwm for many years i love to have clickable (which now i'm using dwmblocks and it has so many bugs/difficulties) status bar, and also i think polybar bloated, its good to think about supporting dwm or xmonad for taking so much popularity to the project,
second, i love to have responsible player support like mpd/moc in status bar
third i have smth in dwm such as tag preview, it is very useful and i think awesome also has smth like that, let me show you a demo..
![Screenshot_2024-07-08-19-38-20_1920x1080](https://github.com/igor-petruk/oatbar/assets/123359596/74b73a28-e01f-4af1-a86c-422bab81cf64)

**Reply by igor-petruk** [reply]
Those are all good features. The approach I was taking with `oatbar` is instead of building a feature directly, I could build a more powerful tool inside of the bar where users can build all these features and more relatively easy.

> for me, since i'm using dwm for many years i love to have clickable (which now i'm using dwmblocks and it has so many bugs/difficulties) status bar

`oatbar` supports `on_mouse_down` command with can run arbitrary command: launch apps, change variables that control the bar itself via `oatctl var`. The question is, which particular actions do you expect in response to those clicks? Is it possible to use third party tool. 

> and also i think polybar bloated

Bloat is a subjective matter expecially when it comes to performance. Something can be minimal and slow, or large, but fast. Another matter is how much dependencies something can have, etc. Did you mean feature bloat? My grievance with `polybar` in particular is how built-in plugins have and "advantage" over custom text. Something I tried to avoid with `oatbar`, where plugins are just purely CLI sources of data and you choose how to display them, all widgets are available. Yet, it is not always possible: you cannot build tray support like that. That must be built-in, unless I demand that users manage it themselves via image blocks, that's not a great experience.

> its good to think about supporting dwm or xmonad for taking so much popularity to the project

Are they not working? Well, I must admit I did not test it on them, maybe there is something that they do different so my panel does not work. I will try to test those.

> i love to have responsible player support like mpd/moc in status bar

Yeah, though about that one. It is possible to build one purely with a bunch of blocks and commands. The question is, how short it looks. I also could build `oatbar-mpd` command which will support track streaming and buttons, and then extend the cookbook on the website with a config snippet to connect it. There probably won't be any first-class mpd block in the oatbar itself, as it would be against it's principles. Do you have examples of the status bar where this is implemented very well? I could prototype with oatbar with shell script extensions only and see how if it looks horrible in config and if some features are need in oatbar to make it easier.

> third i have smth in dwm such as tag preview, it is very useful and i think awesome also has smth like that, let me show you a demo..

Is there a third-party tool I can invoke on hover over the workspace block? It is a great feature to have, but should it be the feature of `oatbar` itself? For example currently if you click on the window title and you have a default config, it would simply call `rofi -show window` and it works seamlessly.

**Reply by Dan7h3x** [reply]
I love the idea of `oatbar-mpd` and i hope so many thing like that, i mean functionality of bar must be high at the user's hand. I am sorry, i hope learn Rust someday and contribute with you to improve these materials, i am having ideas of creating a dynamic tiling DE with full support of ricing and improved functionalities like vim/nvim.
